jobs/payment-jobs/tasks/eft_task.py

```diff
@@ -248,8 +249,29 @@ def _create_receipt_and_invoice(cls,
         if not (invoice_reference := InvoiceReferenceModel.find_by_invoice_id_and_status(
             cil_rollup.invoice_id, InvoiceReferenceStatus.ACTIVE.value
         )):
-            raise Exception(f'Active Invoice reference not '  # pylint: disable=broad-exception-raised
-                            f'found for invoice id: {invoice.id}')
+            raise LookupError(f'Active Invoice reference not '
+                              f'found for invoice id: {invoice.id}')
+        if invoice_reference.is_consolidated:
+            original_invoice_reference = InvoiceReferenceModel.find_by_invoice_id_and_status(
+                cil_rollup.invoice_id, InvoiceReferenceStatus.CANCELLED.value, exclude_consolidated=True
+            )
+            if not original_invoice_reference:
+                raise LookupError(f'Non consolidated cancelled invoice reference not '
+                                  f'found for invoice id: {invoice.id}')
+            invoice_response = CFSService.get_invoice(cfs_account=cfs_account,
+                                                      inv_number=original_invoice_reference.invoice_number)
+            cfs_total = Decimal(invoice_response.get('total', '0'))
+            invoice_total_matches = cfs_total == invoice.total
+            if not invoice_total_matches:
+                raise ValueError(f'SBC-PAY Invoice total {invoice.total} does not match CFS total {cfs_total}')
+            # Note we do the opposite of this in payment_account.
+            current_app.logger.info(f'Consolidated invoice found, reversing consolidated '
+                                    f'invoice {invoice_reference.invoice_number}.')
+            CFSService.reverse_invoice(invoice_reference.invoice_number)
```

Comment:
Should probably use the same code from unlock, I think this is singular, the other route handles multiples